### ISSUE TITLE
helm install abort with  Exception at <.Capabilities.APIVersions.Has>: nil pointer evaluating interface {}.APIVersions

### ISSUE THREAD

**Issue by black-empower** [helm/helm#9471]
Hi,

I am trying to install a chart I created and which evaluates fine using "helm template" on our kubernetes test server.
But when I try to install the chart the Capabilities do not seem to be set and I get the following error:

Error: template: empower/charts/discoveryapi/templates/ingress.yaml:40:34: executing "empower/charts/discoveryapi/templates/ingress.yaml" at <.Capabilities.APIVersions.Has>: nil pointer evaluating interface {}.APIVersions

What I am trying to do is to choose the correct ingress api version depending on the capabilities of the server:
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
apiVersion: networking.k8s.io/v1
{{- else -}}
apiVersion: networking.k8s.io/v1beta1
{{- end }}

The command I am using to try the installation is:
sudo helm install -f values.yaml -n empower-dev empower-dev ./empower --dry-run

What could be the problem that it is not working correctly on the server.
Thanks.

Best regards
Christopher Black

Output of `helm version`:
version.BuildInfo{Version:"v3.5.3", GitCommit:"041ce5a2c17a58be0fcd5f5e16fb3e7e95fea622", GitTreeState:"dirty", GoVersion:"go1.15.8"}

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:52:24Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): selfhosted kubelet on ubuntu 18.04

**Comment by bacongobbler** [comment]
This can occur if you're invoking this `if` clause within a `for` loop or from within another context. Go's context of the dot operator (`.`) changes scope in these blocks.

https://golang.org/pkg/text/template/

One way to work around this is to try using the global operator (`$`). That will allow you to reference the root scope from anywhere EXCEPT in `define` blocks, as those are separate function calls where the root scope changes to the input of the function call.

In other words try this:

```
$.Capabilities.APIVersions.Has
```

**Comment by black-empower** [comment]
Yes I was using it in a for loop.
But if I change that part to the global values other values can no longer be accessed.
It works fine with the default values defined for the chart but as soon as I use the values.yaml (which I also use for a version without capabilities check, where it works fine) an new error occurs:

"YAML parse error on empower/charts/discoveryapi/templates/ingress.yaml: error converting YAML to JSON: yaml: line 19: mapping values are not allowed in this context".
The line number matches the first line after "spec:"
 `{{- if .Values.ingress.tls }}`

Here is my current yaml template:

`{{- if .Values.ingress.enabled -}}
{{- $fullName := include "discoveryapi.fullname" . -}}
{{- $svcPort := .Values.service.port -}}
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
apiVersion: networking.k8s.io/v1
{{- else -}}
apiVersion: networking.k8s.io/v1beta1
{{- end }}
kind: Ingress
metadata:
  name: {{ $fullName }}
  labels:
    {{- include "discoveryapi.labels" . | nindent 4 }}
  {{- with .Values.ingress.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
spec:
  {{- if .Values.ingress.tls }}
  tls:
    {{- range .Values.ingress.tls }}
    - hosts:
        {{- range .hosts }}
        - {{ . | quote }}
        {{- end }}
      secretName: {{ .secretName }}
    {{- end }}
  {{- end }}
  rules:
    {{- range .Values.ingress.hosts }}
    - host: {{ .host | quote }}
      http:
        paths:
          {{- range .paths }}
          - path: {{ . }}
            pathType: Prefix
            backend:
              {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
              service:
                name: {{ $fullName }}
                port: {{ $svcPort }}
              {{- else -}}
              serviceName: {{ $fullName }}
              servicePort: {{ $svcPort }}
              {{- end }}
          {{- end }}
    {{- end }}
{{- end }}
`

What could be the problem.
